osc-server/src/main/java/org/osc/core/broker/service/DSConformService.java

```diff
@@ -0,0 +1,200 @@
+import org.osc.core.broker.job.JobEngine;
+import org.osc.core.broker.job.JobQueuer;
+import org.osc.core.broker.job.JobQueuer.JobRequest;
+import org.osc.core.broker.job.TaskGraph;
+import org.osc.core.broker.job.lock.LockObjectReference;
+import org.osc.core.broker.model.entities.appliance.DistributedAppliance;
+import org.osc.core.broker.model.entities.job.JobRecord;
+import org.osc.core.broker.model.entities.virtualization.VirtualizationConnector;
+import org.osc.core.broker.model.entities.virtualization.openstack.DeploymentSpec;
+import org.osc.core.broker.rest.client.openstack.openstack4j.Endpoint;
+import org.osc.core.broker.service.persistence.DeploymentSpecEntityMgr;
+import org.osc.core.broker.service.persistence.OSCEntityManager;
+import org.osc.core.broker.service.tasks.conformance.UnlockObjectMetaTask;
+import org.osc.core.broker.service.tasks.conformance.k8s.deploymentspec.ConformK8sDeploymentSpecMetaTask;
+import org.osc.core.broker.service.tasks.conformance.openstack.deploymentspec.DSConformanceCheckMetaTask;
+import org.osc.core.broker.util.TransactionalBroadcastUtil;
+import org.osc.core.broker.util.db.DBConnectionManager;
+import org.osc.core.common.job.TaskGuard;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.transaction.control.ScopedWorkException;
+import org.osgi.service.transaction.control.TransactionControl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Component(service = {DSConformService.class})
+public class DSConformService {
+
+    private static final Logger log = LoggerFactory.getLogger(DSConformService.class);
+
+    private final AtomicBoolean initDone = new AtomicBoolean();
+
+    private BundleContext context;
+
+    @Reference
+    protected DBConnectionManager dbConnectionManager;
+
+    @Reference
+    protected TransactionalBroadcastUtil txBroadcastUtil;
+
+    private void delayedInit() {
+        if (this.initDone.compareAndSet(false, true)) {
+            this.dsConformanceCheckMetaTask = this.context.getService(this.dsConformanceCheckMetaTaskSR);
+        }
+    }
+
+    @Activate
+    private void activate(BundleContext context) {
+        this.context = context;
+    }
+
+    @Deactivate
+    private void deactivate(BundleContext context) {
+        if (this.initDone.get()) {
+            context.ungetService(this.dsConformanceCheckMetaTaskSR);
+        }
+    }
+
+    @Reference
+    private ConformK8sDeploymentSpecMetaTask conformK8sDeploymentSpecMetaTask;
+
+    // optional+dynamic to resolve circular reference
```

Comment:
please explain circular reference here..with this refactoring this should be resolved